packages/nimble-components/src/table-column/text-base/group-header-view/index.ts

```diff
@@ -15,7 +19,44 @@ export abstract class TableColumnTextGroupHeaderViewBase<
 
     /**
      * Text to render in the cell.
+     *
+     * The value is initialized to `tableGroupRowPlaceholderNoValue` because if the group
+     * row never has a value defined on it, the change handlers may never get called but
+     * the text needs to be correct.
      */
     @observable
-    public text = '';
+    public text = tableGroupRowPlaceholderNoValueLabel.getValueFor(this);
+
+    protected columnConfigChanged(): void {
+        if (!this.applyPlaceholderTextIfNeeded()) {
+            this.updateText();
+        }
+    }
+
+    protected groupHeaderValueChanged(): void {
+        if (!this.applyPlaceholderTextIfNeeded()) {
+            this.updateText();
+        }
+    }
+
+    protected abstract updateText(): void;
+
+    /**
+     * Sets `this.text` to the appropriate placeholder if `groupHeaderValue` warrants it.
+     * @returns `true` if `this.text` was set to a placeholder, `false` otherwise.
+     */
+    private applyPlaceholderTextIfNeeded(): boolean {
+        if (
+            this.groupHeaderValue === null
+            || this.groupHeaderValue === undefined
+        ) {
+            this.text = tableGroupRowPlaceholderNoValueLabel.getValueFor(this);
```

Comment:
when a token is referenced in the binding of a template then I believe the binding will be updated when the token value changes. We could either move this to the template or subscribe to the tokens and observe them manually.

Doing it in the template may be a little tricky mixing with hasOverflow, didn't play with it.

Should probably also test the group header element responding to label-provider updates. Could either have tests for base specifically or put it in group-header-view-text and call it covered